core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -896,7 +896,14 @@
       "The sampling percentage to use for replication traces"),
   REPLICATION_RPC_TIMEOUT("replication.rpc.timeout", "2m", PropertyType.TIMEDURATION,
       "Amount of time for a single replication RPC call to last before failing"
-          + " the attempt. See replication.work.attempts.");
+          + " the attempt. See replication.work.attempts."),
+  VOLUME_CHOOSER_UPDATE_INTERVAL("general.volume.chooser.update.interval", "2h",
```

Comment:
I think it would be much simpler to just choose on every new file created... rather than make any decision sticky for any amount of time.

Reply:
I am in favor of calling the chooser for each tablet file.  Does anyone know of any performance implications for having all of a tablets files on a single NN versus multiple NNs?  It seems like it should not matter, but I am not completely sure about this.

I think this time limit is not ideal in that it introduces behavior outside of the volume choosers control (like volumes being sticky for tablets).  I think the best way to go is to provide the volume chooser with enough information so that it can make these type of decisions itself.  For example, one possible way to have a sticky chooser would be to provide choosers with the last choice it made for a resource.  Another possible way would be to provide the tablets current set of files to the chooser.  I don't think we need to modify the chooser to provide more info at this time, unless we have a valid use case.  I am not even sure sticky tablets is a good goal.

We could call chooser for each tablet file now.  If later we find this is bad, then we could provide more info to the chooser so it can make better decisions.

Reply:
I thought of an implication for multiple namenodes per tablet.  If tablets are sticky, then a tablets probability of suffering due to a degraded NN is lower.   If there are 5 NN  with each tablet using a single NN then 1 degraded NN will impact 20% of tablets.  However if every tablet has a file on each NN, then a degraded NN will impact 100% of tablets.

If a system is currently using bulk import, then it may already have multiple NN per tablet.

Reply:
Choosers should be given enough information so they have the option of selecting a preferred set of volumes based on previous choices for the same tablet... so they can avoid such scenarios. Currently, I believe choosers get to see the tabletId or KeyExtent